drivers/i2c/i2c_ite_enhance.c

```diff
@@ -0,0 +1,771 @@
+/*
+ * Copyright (c) 2022 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_enhance_i2c
+
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <drivers/pinmux.h>
+#include <errno.h>
+#include <logging/log.h>
+LOG_MODULE_REGISTER(i2c_ite_enhance, CONFIG_I2C_LOG_LEVEL);
+#include "i2c-priv.h"
+#include <soc.h>
+#include <soc_dt.h>
+#include <sys/util.h>
+
+/* Start smbus session from idle state */
+#define I2C_MSG_START BIT(5)
+
+#define I2C_LINE_SCL_HIGH BIT(0)
+#define I2C_LINE_SDA_HIGH BIT(1)
+#define I2C_LINE_IDLE (I2C_LINE_SCL_HIGH | I2C_LINE_SDA_HIGH)
+
+/*
+ * Structure i2c_alts_cfg is about the alternate function
+ * setting of i2c, this config will be used at initial
+ * time and recover bus.
+ */
+struct i2c_alts_cfg {
+	/* Pinmux control group */
+	const struct device *pinctrls;
```

Comment:
gentle reminder: you should implement a pinctrl driver.

Reply:
Pinctrl driver is under development.